ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_linalg.py

```diff
@@ -1611,3 +1611,54 @@ def test_tucker_tensorly(tol_norm_2, tol_max_abs, shape, ranks):
         ivy.max(ivy.abs(rec_svd - rec_random)) < tol_max_abs,
         "abs norm of difference between svd and random init too high",
     )
+
+
+# dot
+@st.composite
+def _generate_dot_dtype_and_arrays(draw):
+    matrices_dims = draw(
+        st.lists(st.integers(min_value=2, max_value=10), min_size=3, max_size=3)
+    )
+    shape_1 = (matrices_dims[0], matrices_dims[1])
+    shape_2 = (matrices_dims[1], matrices_dims[2])
+
+    dtype_1, matrix_1 = draw(
+        helpers.dtype_and_values(
+            shape=shape_1,
+            available_dtypes=helpers.get_dtypes("float"),
+            min_value=-10,
+            max_value=10,
+        ).filter(lambda x: "float16" not in x[0])
+    )
+    dtype_2, matrix_2 = draw(
+        helpers.dtype_and_values(
+            shape=shape_2,
+            dtype=dtype_1,
+            min_value=-10,
+            max_value=10,
+        ).filter(lambda x: "float16" not in x[0])
+    )
```

Comment:
As discussed, we shouldn't only be generating two dimensional tensors in our tests but rather when the dimensions of `a` and `b` are equal to 2, then we should make sure that `a` and `b` are matmul compatible. In general, if you look at the [documentation ](https://numpy.org/doc/stable/reference/generated/numpy.dot.html)of `np.dot`, you'd see that there always needs to be one dimension that should have the same size in both `a` and `b`.  Specifically,

> If both a and b are 1-D arrays, it is inner product of vectors --> dim 0 should be equal
> 
> If both a and b are 2-D arrays, it is matrix multiplication --> dim1(a) == dim0(b) 

> If either a or b is 0-D (scalar), it is equivalent to multiply -- no constraints.
> 
> If a is an N-D array and b is a 1-D array, it is a sum product over the last axis of a and b --> size(b) == size of last dimension of a. 
> 
> If a is an N-D array and b is an M-D array (where M>=2), it is a sum product over the last axis of a and the second-to-last axis of b --> size of last dim of a === size of penultimate dim of b.

So, you should be doing the following in the test :
```
generate a shape for a = shape_a

generate a shape for b = shape_b

generate common dimension size = contract_dim

based on the conditions mentioned above, update the size of the common dimension

generate a and b
```